sentry_sdk/tracing.py

```diff
@@ -469,6 +475,10 @@ def __init__(
         Span.__init__(self, **kwargs)
         self.name = name
 
+        # transactions (when considered as spans) belong to themselves (when
+        # considered as transactions)
+        self._containing_transaction = self
```

Comment:
would much rather store `None` or some other special value than create circular references

Reply:
What kind of a special value were you thinking? I suppose it could be `None`, as long as the getting does a typecheck first. Does that seem like a better option?

Reply:
```python
    @property
    def transaction(self):
        # type: () -> Any
        # would be type: () -> Optional[Transaction], see https://github.com/python/mypy/issues/3004
        """Return the transaction (root span) in the scope, if any."""

        # there is no span/transaction on the scope
        if self._span is None:
            return None

        # the span on the scope is itself a transaction
        if isinstance(self._span, Transaction):
            return self._span

        # the span on the scope isn't a transaction but belongs to one
        if self._span._containing_transaction:
            return self._span._containing_transaction

        # there's a span (not a transaction) on the scope, but it was started on
        # its own, not as the descendant of a transaction (this is deprecated
        # behavior, but as long as the start_span function exists, it can still
        # happen)
        return None
```

Reply:
(Out of curiosity, is there a drawback to the circular reference?)

Reply:
I am not sure if `None` is already used as a special value. If it is we need another one ("self" or something else). But yeah in either case the getter needs to special-case.

Reply:
Hmmm - that also means `start_child` is going to have to special-case, because the attribute's value can't just be passed from parent to child.

Can you help me understand the disadvantages of the circular reference? It seems a lot cleaner.

Reply:
creating circular references is generally less easy on memory as they are collected whenever, as opposed to "linear" ones, and the gc is also slower if it needs to do more.

don't ask me by how much or if it matters, but if it's avoidable it's certainly better. I would create a new sentinel value like this:

```
_SELF = object()
```

and check for it like this:

```
if self._containing_transaction is _SELF:
```

Reply:
Ah, okay. Makes sense, thanks.

Sentinel better than typechecking, you think?

Reply:
Ah no I don't think that matters

Reply:
👍🏻

Reply:
Okay, fixed in https://github.com/getsentry/sentry-python/pull/870. (Rodolfo has been coaching me to push smaller PRs, but I might have gone overboard a little...)

Once that's approved and merged, I'll push the getter changes.